src/components/TreeView.js

```diff
@@ -150,15 +147,24 @@ function formatSignature(node, separator) {
   return <span style={style}><strong>{node.type}</strong>{separator}{node.value.value} {symbol}</span>
 }
 
-// Signature fetch may exist in three different states so it requires 
+// Signature fetch may exist in different states so it requires
 // richer formating with ternary based coloring based on whether the
-// state is valid (VALID), is pending (NOT_VERIFIED_YET), or has errored (INVALID).
+// state is NONE, SUCCCESS, PENDING, FAIL or NOT_EXIST.
 function formatSignatureCheckState(node, separator) {
   let message = "";
-  if (node.state === SIGNATURE.INVALID) {
-    message = <span style={{color: "red"}}> Error checking signatures...</span>;
-  } else if (node.state === SIGNATURE.NOT_VERIFIED_YET) {
-    message = <span style={{fontStyle: "italic"}}> Checking signatures...</span>;
+  switch (node.state) {
```

Comment:
You don't have to change this, but just a suggestion for the future. `state` is a term of art in React, so I avoid using that as the name. It might have been clearer to name this prop 'signatureCheckState' or something.

Reply:
https://github.com/stellar/laboratory/issues/367